### DISCUSSION TITLE
Need advice for updating localization

### DISCUSSION THREAD

**Discussion by FRtranslator** [utmapp/UTM#4886] [General]
Hi,

I'm quite a newbie in coding, especially with Xcode.
Now that I've done all the translation up to 4.1.1, I would like to know how can I find the updated / added / removed strings so I can update the (French) translation.
Detailed "how to" is much welcome (by other translators or anyone).

Since now I updated my translation based on other languages differences/updates, but now I'd like to do it the right way, by generating updates from the project itself.

Thanks for your help!

**Comment by osy** [comment]
Adding some other translators: @MMP0 @JezerM

**Comment by MMP0** [comment]
You can export localizable files in the project at `Product` > `Export Localizations...`, but I would not recommend it as its accuracy is very poor.
We have to go through all the files and keep an eye on the strings added to each commit.

@osy, if possible, it would be nice if you could add the strings to `en.lproj/Localizable.strings` as well when adding them to the code. It will make it easier to track changes.

**Reply by osy** [reply]
I have to admit I don't really know how translations work myself so please open an issue with suggestions like that. I was also wondering if it is better to use placeholder text in the code such as "VM_CANNOT_START_ERROR_MESSAGE"?

**Reply by MMP0** [reply]
Placeholders seem to be deprecated in SwiftUI according to Apple. The original string corresponds to the key.
Additionally, `LocalizedStringKey` in SwiftUI overrides string interpolation, e.g. `Text("\(name) Drive")` automatically references the format string `"%@ Drive"`.

**Comment by JezerM** [comment]
Well, I don't use specialized software for localizations, as the english one doesn't cover all the strings in UTM. So, a `Export Localizations...` just does not work.

I used the French and Japanese localizations to add my own for Spanish, looking one by one. If there is a new update, I check the Japanese localization as it is the most up-to-date, or either I try UTM to check where I could translate.

@osy, I also agree with @MMP0 about English localization. Whenever possible, try to add your string changes in the corresponding files, as it would make localization much easier.

**Comment by FRtranslator** [comment]
I see, there is no easy way to update translations.
Reading the entire file(s) to check for any modification/update is a mess :(

There is no way to get this easier, or is Xcode the limit?

**Reply by osy** [reply]
I will try to improve the situation in the next version with people’s feedback.
